### ISSUE TITLE
Windows: Arrow keys don't work in vim

### ISSUE THREAD

**Issue by davidhewitt** [jwilm/alacritty#2141]
Arrow keys don't function in tools installable with git for windows (e.g. `C:\Program Files\Git\usr\bin\vim.exe`).  I think they're sending some kind of input but it's not being recieved correctly. In `less` if I hold down the keys I see e.g. `Esc[A`, `Esc[B` flicker repeatedly in the bottom-left corner of the screen.

These tools do work correctly in cmd, but not in Alacritty (winpty or conpty). I think they might have used to work in Alacritty too, but broke sometime around the last glutin upgrade.

A related feature I'd really like to see - at the moment the mousewheel scrolls Alacritty's scrollback rather than scrolling the buffer of `less`. Perhaps should I open a separate issue for that?

I'm happy to spend some time digging into these things. Any suggestions on where I should start?

**Comment by chrisduerr** [comment]
Can this issue be reproduced with both WinPTY and ConPTY?

Also scrolling the buffer of `less` is already supported on Linux, this is likely a Windows bug. Though there is no fool-proof way to do it, so it's only possible to do it as a heuristic that's bound to get into some people's way (like users of `tmux`, it's impossible to do it there).

**Comment by davidhewitt** [comment]
> Can this issue be reproduced with both WinPTY and ConPTY?

Yep. 😢 

> Also scrolling the buffer of `less` is already supported on Linux, this is likely a Windows bug.

Nice! I'll take a look and see if I can figure out how it's working on the linux side (and how I might be able to fix windows support)...

**Comment by davidhewitt** [comment]
Just a follow up on these, looks like it's a deficiency in Windows. The relevant escape sequences for both of these features (arrow keys in vim / cursor keys) and scrolling in less (alternate screen buffer) are don't seem to ever be sent to Alacritty.

I think "fixing" this in Alacritty requires no more than waiting for support to be added by Microsoft to the ConPTY interface. (I'm not sure whether this is something that Winpty could ever support.)

**Comment by frostgaming** [comment]
@davidhewitt I ran into this issue a while ago and was able to fix it by adding this to my ~/.vimrc for Cygwin.
```
:set term=builtin_ansi
```

Arrowkeys worked fine after that. Hope it helps you guys.

**Comment by davidhewitt** [comment]
Thanks for the tip - sadly doesn't seem to help in the cmd.exe case :(

**Comment by frostgaming** [comment]
@davidhewitt Does adding this to the bottom of `C:\Program Files\Git\etc\vimrc` fix your arrowkeys for cmd.exe?
```
set <Up>=[A
set <Down>=[B
set <Right>=[C
set <Left>=[D
```

**Comment by davidhewitt** [comment]
Yes, it did! Inspired by that I also made the following input file to [lesskey](https://linux.die.net/man/1/lesskey) to fix keybinds in less:

    \e[A back-line
    \e[B forw-line

Between those two I now have working arrow keys everywhere 😄 

It would still be nice for scrollback in less to work "properly" - but as described above that I believe is waiting on Microsoft to add support to ConPTY. So I'm going to close this ticket as resolved.

Thanks!
